### ISSUE TITLE
column width too short for "name"

### ISSUE THREAD

**Issue by gessel** [owncloud/music#632]
A (probably misformatted) music file with a rather long name generates an error that stops the scan with:

Doctrine\DBAL\Exception\DriverException: An exception occurred while executing 'INSERT INTO `oc_music_artists`(`name`,`user_id`,`hash`) VALUES(?,?,?)' with params [" (very long title here)", "gessel", "27deb919402deae1266c3643001ee235"]: SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'name' at row 1

It appears to hang the scan - truncating the name would be fine.

**Comment by paulijar** [comment]
Thanks for the report. Roughly how long is your "very long title". Is it hundreds or thousands of characters? Way earlier, there was a report that names over 256 characters would cause some problem but I couldn't reproduce the problems then with 260 characters long name, on my system those got truncated properly. Which database system are you using?

**Comment by gessel** [comment]
Hi Pauli!

My database is MySQL, using redis for data caching.  The text is 281 characters.  It is obviously a mistake - description, not title in the Mp3 meta info, but...  


Here's the full error:

Doctrine\DBAL\Exception\DriverException: An exception occurred while executing 'INSERT INTO `oc_music_artists`(`name`,`user_id`,`hash`) VALUES(?,?,?)' with params ["Angela Gheorghiu (Artist), Roberto Alagna (Artist), Daniela Barcellona (Artist), Julian Konstantinov (Artist), Giuseppe Verdi (Composer), Claudio Abbado (Conductor), Berlin Philharmonic Orchestra (Orchestra), Swedish Radio Chorus (Performer), Eric Ericson Chamber Choir (Performer)", "gessel", "27deb919402deae1266c3643001ee235"]: SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'name' at row 1

    /usr/local/www/nextcloud/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/DBALException.php - line 128: Doctrine\DBAL\Driver\AbstractMySQLDriver->convertException('An exception oc...', Object(Doctrine\DBAL\Driver\PDOException))
    /usr/local/www/nextcloud/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Statement.php - line 177: Doctrine\DBAL\DBALException driverExceptionDuringQuery(Object(Doctrine\DBAL\Driver\PDOMySql\Driver), Object(Doctrine\DBAL\Driver\PDOException), 'INSERT INTO `oc...', Array)
    /usr/local/www/nextcloud/lib/public/AppFramework/Db/Mapper.php - line 247: Doctrine\DBAL\Statement->execute()
    /usr/local/www/nextcloud/lib/public/AppFramework/Db/Mapper.php - line 125: OCP\AppFramework\Db\Mapper->execute('INSERT INTO `*P...', Array)
    /usr/local/www/nextcloud/apps/music/db/basemapper.php - line 96: OCP\AppFramework\Db\Mapper->insert(Object(OCA\Music\Db\Artist))
    /usr/local/www/nextcloud/apps/music/businesslayer/artistbusinesslayer.php - line 52: OCA\Music\Db\BaseMapper->insertOrUpdate(Object(OCA\Music\Db\Artist))
    /usr/local/www/nextcloud/apps/music/utility/scanner.php - line 147: OCA\Music\BusinessLayer\ArtistBusinessLayer->addOrUpdateArtist('Angela Gheorghi...', 'gessel')
    /usr/local/www/nextcloud/apps/music/utility/scanner.php - line 115: OCA\Music\Utility\Scanner->updateAudio(Object(OC\Files\Node\File), 'gessel', Object(OC\Files\Node\Folder), '/gessel/files/M...', 'audio/mpeg')
    /usr/local/www/nextcloud/apps/music/utility/scanner.php - line 428: OCA\Music\Utility\Scanner->update(Object(OC\Files\Node\File), 'gessel', Object(OC\Files\Node\Folder))
    /usr/local/www/nextcloud/apps/music/controller/apicontroller.php - line 389: OCA\Music\Utility\Scanner->scanFiles('gessel', Object(OC\Files\Node\Folder), Array)
    [internal function] OCA\Music\Controller\ApiController->scan('60085,59090,465...', false)
    /usr/local/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php - line 161: call_user_func_array(Array, Array)
    /usr/local/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php - line 91: OC\AppFramework\Http\Dispatcher->executeController(Object(OCA\Music\Controller\ApiController), 'scan')
    /usr/local/www/nextcloud/lib/private/AppFramework/App.php - line 115: OC\AppFramework\Http\Dispatcher->dispatch(Object(OCA\Music\Controller\ApiController), 'scan')
    /usr/local/www/nextcloud/lib/private/AppFramework/Routing/RouteActionHandler.php - line 47: OC\AppFramework\App main('ApiController', 'scan', Object(OC\AppFramework\DependencyInjection\DIContainer), Array)
    [internal function] OC\AppFramework\Routing\RouteActionHandler->__invoke(Array)
    /usr/local/www/nextcloud/lib/private/Route/Router.php - line 297: call_user_func(Object(OC\AppFramework\Routing\RouteActionHandler), Array)
    /usr/local/www/nextcloud/lib/base.php - line 998: OC\Route\Router->match('/apps/music/api...')
    /usr/local/www/nextcloud/index.php - line 37: OC handleRequest()
    {main}



-------- Original Message --------
Subject: Re: [owncloud/music] column width too short for "name" (#632)

**Comment by paulijar** [comment]
Thanks. I tried to set artist name of one track to the one shown in your error message but I still couldn't reproduce the issue. The file got scanned and the track was shown on the UI with artist name truncated to 256 characters. I'm not sure, though, which layer did the truncation. There's no such logic in the Music app, so it was either the ownCloud core or the database management system (MySQL). I tested on ownCloud 9.1 and Nextcloud 13.

Which version of ownCloud/Nextcloud and MySQL are you using.

**Comment by gessel** [comment]
I'll send you the file when i get home.

**Comment by gessel** [comment]
File (should) be at:
https://nas.blackrosetech.com/index.php/s/oRJfeGnfF5E52C5

nextcloud-13.0.0
mysql56-client-5.6.39_1     

mysql56-server-5.6.39_1
pecl-redis-3.1.6
redis-3.2.11
FreeBSD 10.3-RELEASE-p26 #0 r327588:

The parameter is "Album Artist" (I use MP3Tag on windows for this sort of stuff, lovely program)

Thanks!


(full error reattached just for completeness)


Doctrine\DBAL\Exception\DriverException: An exception occurred while executing 'INSERT INTO `oc_music_artists`(`name`,`user_id`,`hash`) VALUES(?,?,?)' with params ["Angela Gheorghiu (Artist), Roberto Alagna (Artist), Daniela Barcellona (Artist), Julian Konstantinov (Artist), Giuseppe Verdi (Composer), Claudio Abbado (Conductor), Berlin Philharmonic Orchestra (Orchestra), Swedish Radio Chorus (Performer), Eric Ericson Chamber Choir (Performer)", "gessel", "27deb919402deae1266c3643001ee235"]: SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'name' at row 1

    /usr/local/www/nextcloud/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/DBALException.php - line 128: Doctrine\DBAL\Driver\AbstractMySQLDriver->convertException('An exception oc...', Object(Doctrine\DBAL\Driver\PDOException))
    /usr/local/www/nextcloud/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Statement.php - line 177: Doctrine\DBAL\DBALException driverExceptionDuringQuery(Object(Doctrine\DBAL\Driver\PDOMySql\Driver), Object(Doctrine\DBAL\Driver\PDOException), 'INSERT INTO `oc...', Array)
    /usr/local/www/nextcloud/lib/public/AppFramework/Db/Mapper.php - line 247: Doctrine\DBAL\Statement->execute()
    /usr/local/www/nextcloud/lib/public/AppFramework/Db/Mapper.php - line 125: OCP\AppFramework\Db\Mapper->execute('INSERT INTO `*P...', Array)
    /usr/local/www/nextcloud/apps/music/db/basemapper.php - line 96: OCP\AppFramework\Db\Mapper->insert(Object(OCA\Music\Db\Artist))
    /usr/local/www/nextcloud/apps/music/businesslayer/artistbusinesslayer.php - line 52: OCA\Music\Db\BaseMapper->insertOrUpdate(Object(OCA\Music\Db\Artist))
    /usr/local/www/nextcloud/apps/music/utility/scanner.php - line 147: OCA\Music\BusinessLayer\ArtistBusinessLayer->addOrUpdateArtist('Angela Gheorghi...', 'gessel')
    /usr/local/www/nextcloud/apps/music/utility/scanner.php - line 115: OCA\Music\Utility\Scanner->updateAudio(Object(OC\Files\Node\File), 'gessel', Object(OC\Files\Node\Folder), '/gessel/files/M...', 'audio/mpeg')
    /usr/local/www/nextcloud/apps/music/utility/scanner.php - line 428: OCA\Music\Utility\Scanner->update(Object(OC\Files\Node\File), 'gessel', Object(OC\Files\Node\Folder))
    /usr/local/www/nextcloud/apps/music/controller/apicontroller.php - line 389: OCA\Music\Utility\Scanner->scanFiles('gessel', Object(OC\Files\Node\Folder), Array)
    [internal function] OCA\Music\Controller\ApiController->scan('60085,59090,465...', false)
    /usr/local/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php - line 161: call_user_func_array(Array, Array)
    /usr/local/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php - line 91: OC\AppFramework\Http\Dispatcher->executeController(Object(OCA\Music\Controller\ApiController), 'scan')
    /usr/local/www/nextcloud/lib/private/AppFramework/App.php - line 115: OC\AppFramework\Http\Dispatcher->dispatch(Object(OCA\Music\Controller\ApiController), 'scan')
    /usr/local/www/nextcloud/lib/private/AppFramework/Routing/RouteActionHandler.php - line 47: OC\AppFramework\App main('ApiController', 'scan', Object(OC\AppFramework\DependencyInjection\DIContainer), Array)
    [internal function] OC\AppFramework\Routing\RouteActionHandler->__invoke(Array)
    /usr/local/www/nextcloud/lib/private/Route/Router.php - line 297: call_user_func(Object(OC\AppFramework\Routing\RouteActionHandler), Array)
    /usr/local/www/nextcloud/lib/base.php - line 998: OC\Route\Router->match('/apps/music/api...')
    /usr/local/www/nextcloud/index.php - line 37: OC handleRequest()
    {main}


-------- Original Message --------
Subject: Re: [owncloud/music] column width too short for "name" (#632)

**Comment by paulijar** [comment]
I see only a playlist file (m3u) in your link. Please send the file (or correct link) directly to my email address instead of sending through github. You can find my email address from my github profile.

**Comment by paulijar** [comment]
Okay, I got your file now. However, it did not cause any problem on my system. My MySQL version is slightly different (5.5.55) and of course a server setup consists of numerous components and their configurations.

I guess I just have to add some truncations "blindly" and then you can test if it helps. However, now it's getting late and I'm busy tomorrow so this has to wait until later this week.

**Comment by gessel** [comment]
No problem at all - I should just fix the labels to something that fits 255 really.

But I'm happy to leave it and test anything.  Very glad to help in any way I can.

-David

-------- Original Message --------
Subject: Re: [owncloud/music] column width too short for "name" (#632)

**Comment by paulijar** [comment]
Could you try replacing the file apps/music/businesslayer/artistbusinesslayer.php with the one from the attached zip? Of course make a backup of your original file first. Nextcloud will show you a note "There were problems with the code integrity check" (if logged in as admin) because you have tampered with a file which is part of a signed application package but you don't need to worry about it. Afterwards, you can restore the original file and get rid of the note.

[artistbusinesslayer.zip](https://github.com/owncloud/music/files/1790674/artistbusinesslayer.zip)

**Comment by paulijar** [comment]
@gessel Have you been able to test with the provided patch?

**Comment by e-alfred** [comment]
I am facing the same bug:

```
sudo -u www-data php occ  music:scan --all
Start scan for user
Found 161 new music files
Scanning /user/files/SMB server/install/portableapps/PortableApps/ChromiumBSUPortable/App/cbsu/data/wav/power.wav

In AbstractMySQLDriver.php line 115:
                                                                                                                                                                                                                                       
  An exception occurred while executing 'INSERT INTO `oc_music_tracks`(`title`,`artist_id`,`album_id`,`file_id`,`mimetype`,`user_id`,`length`,`bitrate`) VALUES(?,?,?,?,?,?,?,?)' with params ["(\u0000\u0000\u0000\u001f\u0000\u0000  
  \u0000\u0015\u0000\u0000\u0000\u0001\u0000\b\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0  
  000\u00bf\u0000\u0000\u00bf\u0000\u0000\u0000\u00bf\u00bf\u0000\u00bf\u0000\u0000\u0000\u00bf\u0000\u00bf\u0000\u00bf\u00bf\u0000\u0000\u00c0\u00c0\u00c0\u0000\u00c0\u00dc\u00c0\u0000\u00f0\u00ca\u00a6\u0000\u0000\u0000\u0000\u  
  0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\  
  u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000  
  \u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u000  
  0\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u00  
  00\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0  
  000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u  
  0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\  
  u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000  
  \u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u000  
  0\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u00  
  00\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0  
  000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u  
  0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\  
  u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000  
  \u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u000  
  0\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u00  
  00\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0  
  000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u  
  0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\  
  u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000  
  \u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u000  
  0\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u00  
  00\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0  
  000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u  
  0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u00f0\u00fb\u00ff\u0000\u00a4\  
  u00a0\u00a0\u0000\u0080\u0080\u0080\u0000\u0000\u0000\u00ff\u0000\u0000\u00ff\u0000\u0000\u0000\u00ff\u00ff\u0000\u00ff\u0000\u0000\u0000\u00ff\u0000\u00ff\u0000\u00ff\u00ff\u0000\u0000\u00ff\u00ff\u00ff\u0000\u00ff\u0000\u0000  
  \u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u00ff\\\u0000\u00f8\u00f8\u00f8\u00f8\u0000\u00f8\u00f8\u00f8\u0  
  000\u0000\u0000\u00f8\u00f8\u0000\u0000\u0000\u00f8\u00f8\u0000\u0000\u0000\u00f8\u00f8\u0000\u00f8\u00f8\u00f8\u00f8\u00f8\u0000\\\u0000\u00f8\u00f8\u00f8\u00f8\u0000\u00f8\u00f8\u00f8\u00f8\u00f8\u00f8\u00f8\u00f8\u00f8\u00f8  
  \u00f8\u00f8\u00f8\u00f8\u00f8\u00f8\u00f8\u00f8\u0000\u00f8\u00f8\u00f8\u00f8\u00f8\u0000\\\u0000\u00ff\u0000\u0007\u0007\u00ff\u0007\u0007\u0000\u00ff\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0000\u00ff\u0  
  007\u0007\u0000\u0007\u0007\u0000\u00f8\u00f8\u0000\\\u0000\u00ff\u0007\u0007\u0007\u0007\u00ff\u0007\u0007\u0007\u0000\u00ff\u0007\u0007\u0007\u0007\u0007\u0007\u0000\u00ff\u0007\u0007\u0007\u00f8\u00f8\u0007\u0007\u0007\u00f8  
  \u00f8\u0000\\\u0000\u00ff\u0007\u0007\u0007\u0007\u00ff\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0000\u00ff\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0000\u0007\u0007\u0007\u0007\u00f8\u00f8\u0000\u0007\u0000\u00ff\u0007\u0  
  007\u0007\u0007\u0007\u00ff\u00ff\u00ff\u00ff\u00ff\u00ff\u00ff\u00ff\u00ff\u00ff\u00ff\u00ff\u00ff\u00ff\u00ff\u00f8\u00f8\u0007\u0007\u0007\u0007\u00f8\u00f8\u0000\u0007\u0000\u00ff\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u  
  0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u00f8\u00f8\u0000\u0007\u0000\u00ff\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\  
  u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u00f8\u00f8\u0000\u0007\u0000\u00ff\u0007\u0007\u0007\u00ff\u00ff\u00ff\u00ff\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007  
  \u0007\u0007\u00ff\u00ff\u00ff\u00ff\u0007\u00f8\u00f8\u0000\u0007\u0000\u00ff\u0007\u0007\u00f8\u0000\u0000\u0000\u00ff\u00ff\u0007\u0000\u00ff\u00ff\u00ff\u00ff\u00ff\u00ff\u00ff\u00ff\u0007\u0007\u00f8\u0000\u0000\u0000\u00f  
  f\u00ff\u00f8\u00f8\u0000\u0007\u0000\u00ff\u0007\u00f8\u0000\u00ff\u0000\u00ff\u0000\u00ff\u0007\u0000\u0003\u00fb\u00fb\u00fb\u0003\u0007\u0007\u00ff\u0007\u00f8\u0000\u00ff\u0000\u00ff\u0000\u00ff\u00f8\u00f8\u0000\u0007\u00  
  00\u00ff\u0007\u00f8\u0000\u0000\u00f8\u0000\u0000\u00ff\u0007\u0000\u0003\u00fb\u00fb\u00fb\u0003\u0007\u0007\u00ff\u0007\u00f8\u0000\u0000\u00f8\u0000\u0000\u00ff\u00f8\u00f8\u0000\u00f8\u0000\u00ff\u0007\u00f8\u0000\u00ff\u0  
  000\u00ff\u0000\u00ff\u0007\u0000\u0003\u0003\u0003\u0003\u0003\u00f8\u00f8\u00ff\u0007\u00f8\u0000\u00ff\u0000\u00ff\u0000\u00ff\u00f8\u00f8\u0000\u0007\u0000\u00ff\u0007\u00f8\u00f8\u0000\u0000\u0000\u00f8\u0007\u0007\u0000\u  
  0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0007\u00f8\u00f8\u0000\u0000\u0000\u00f8\u0007\u00f8\u00f8\u0000\u0007\u0000\u00ff\u0007\u0007\u00f8\u00f8\u00f8\u00f8\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\  
  u0007\u0007\u0007\u0007\u00f8\u00f8\u00f8\u00f8\u0007\u0007\u00f8\u00f8\u0000\u0007\u0000\u00ff\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007  
  \u0007\u0007\u0007\u0007\u00f8\u00f8\u0000\u00ff\u0000\u00ff\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u00f8\u00f  
  8\u0000\u00ff\u0000\u00ff\u0000\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0000\u00f8\u00f8\u0000'\u0000\u00ff\u00ff\u00ff\u0  
  0ff\u00ff\u00ff\u00ff\u00ff\u00ff\u00ff\u00ff\u00ff\u00ff\u00ff\u00ff\u00ff\u00ff\u00ff\u00ff\u00ff\u00ff\u00ff\u00ff\u00ff\u00ff\u00ff\u00ff\u00f8\u00f8\u0000'\u00ff\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\  
  u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u00ff'", 1578, 2892, 1553225, "audio\/wav", "user", 1, 176400]:                                                  
                                                                                                                                                                                                                                       
  SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'title' at row 1                                                                                                                                        
                                                                                                                                                                                                                                       

In PDOStatement.php line 107:
                                                                                                 
  SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'title' at row 1  
                                                                                                 

In PDOStatement.php line 105:
                                                                                                 
  SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'title' at row 1  

```

**Comment by e-alfred** [comment]
@paulijar I tried to replace the artistbusinesslayer.php file but it doesn't work. I get the same error as above.

**Comment by paulijar** [comment]
@e-alfred In your case, patching the `artistbusinesslayer.php` does not help because your error is related to the table `oc_music_tracks` while the original reporter's problem was on the table `oc_music_artists`. To do a similar fix on `oc_music_tracks`, you should modify the file `trackbusinesslayer.php`. Find a function called `addOrUpdateTrack` and replace the line
```
$track->setTitle($title);
```
with
```
$track->setTitle(mb_strimwidth($title, 0, 256));
```

However, the root cause of your problem is now that something weird is going on with your file power.wav. The title of the track seems to consist of 1718 unprintable characters. Either the metadata of the file is actually corrupted or then the metadata is good but the Music app has somehow read it incorrectly. Can you check the metadata with some other application?

**Comment by e-alfred** [comment]
@paulijar Thanks, this fixed the issue with the file. I will have to check the file, but as you can see from the filename it is part of Chromium (portable app).

But now I get a Segfault from an *.m3u file without any further error message.

**Comment by paulijar** [comment]
@e-alfred
> But now I get a Segfault from an *.m3u file without any further error message if I do the scan using occ. Verbose output doesn't give any additional errors either.

There are two issues here now. Firstly, the Music app is not really supposed to scan m3u files but that is a known issue and being tracked in #674. Still, even if the app scans m3u it is not supposed to segfault. Looking closer at your previous error print, it seems that the problem file was residing on a SMB server. Is it the same with the m3u file? I suspect that the our metadata extraction is now somehow misbehaving with your SMB setup. The SMB has proved to be problematic before, but usually the problem has been just that no metadata has been extracted from the files on SMB server (see e.g.  #670).

One of the problems with SMB is that there are so many moving parts: there's the ownCloud/Nextcloud SMB integration, which may have different bugs in different cloud versions. Then there are several alternative Linux SMB clients which are utilized by the former. And then there are different SMB server backends. Some combinations of these will probably work just fine and others will not.

**Comment by e-alfred** [comment]
@paulijar Okay, this seems to be an SMB problem for sure. Is it possible to exclude certain folders from scanning or maybe external storage in general (like the desktop client having an option to exclude sync from external storage). Currently the problem is that I have a popover in the UI telling me that there is "new music available" which simply hangs at 2100 of 2300 files. This is annoying because it doesn't go away at all.

**Comment by paulijar** [comment]
@e-alfred Well, you can select the parent folder where the music is looked from. By default, this is your home folder which contains also all the externally mapped folders. You can find this from the "Settings" view of the Music application UI. But if your music is scattered to multiple sub folders of the home folder, then you are out of luck; there's no option to select multiple top-level folders nor to exclude some sub-folders of the selected parent folder.

**Comment by paulijar** [comment]
In the newly released version 0.11.0, titles of tracks/albums/artists are truncated to the maximum width before inserting them to the database. This should fix the issue reported here (although I never could reproduce the problem myself). Please reopen if such problems still occur.
